app/components/post_attachment_opengraph/post_attachment_opengraph.js

```diff
@@ -372,3 +336,55 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
         },
     };
 });
+
+const getBestImageUrl = (data, imagesMetadata, deviceHeight, deviceWidth, isReplyPost) => {
+    if (!data || !data.images) {
+        return {
+            hasImage: false,
+        };
+    }
+
+    const bestDimensions = {
+        width: getViewPostWidth(deviceHeight, deviceWidth, isReplyPost),
+        height: MAX_IMAGE_HEIGHT,
+    };
+
+    const bestImage = getNearestPoint(bestDimensions, data.images, 'width', 'height');
+    const imageUrl = bestImage.secure_url || bestImage.url;
+
+    let ogImage;
+    if (imagesMetadata && imagesMetadata[imageUrl]) {
+        ogImage = imagesMetadata[imageUrl];
+    }
+
+    if (!ogImage) {
+        ogImage = data.images.find((i) => i.url === imageUrl || i.secure_url === imageUrl);
+    }
+
+    // Fallback when the ogImage does not have dimensions but there is a metaImage defined
+    const metaImages = imagesMetadata ? Object.values(imagesMetadata) : null;
+    if ((!ogImage?.width || !ogImage?.height) && metaImages?.length) {
+        ogImage = metaImages[0];
+    }
+
+    let dimensions = bestDimensions;
+    if (ogImage?.width && ogImage?.height) {
+        dimensions = calculateDimensions(ogImage.height, ogImage.width, getViewPostWidth(deviceHeight, deviceWidth, isReplyPost));
+    }
+
+    FastImage.preload([{uri: imageUrl}]);
+
+    return {
+        hasImage: true,
+        ...dimensions,
+        openGraphImageUrl: imageUrl,
+    };
+};
+
+const getViewPostWidth = (deviceHeight, deviceWidth, isReplyPost) => {
```

Comment:
Similar to `getBestImageUrl`, these arguments always originate from `this.props`, so the function itself could handle without needing to pass in arguments?

Reply:
I have updated this method.